src/components/Panels/ElementsPanel/ElementsPanel.vue

```diff
@@ -0,0 +1,186 @@
+                                    :alt="element.name"
+                                />
+                            </div>
+                        </div>
+                    </v-expansion-panel-text>
+                </v-expansion-panel>
+            </v-expansion-panels>
+            <div
+                id="Help"
+                lines="one"
+                :class="tooltipText != 'null' ? 'show' : ''"
+            >
+                {{ tooltipText }}
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import PanelHeader from '@/Panels/PanelHeader.vue'
+import metadata from '#/simulator/src/metadata.json'
+import simulationArea from '#/simulator/src/simulationArea'
+import { uxvar } from '#/simulator/src/ux'
+import modules from '#/simulator/src/modules'
+import { onBeforeMount, ref } from 'vue'
+var panelData = []
+window.elementHierarchy = metadata.elementHierarchy
+window.elementPanelList = []
+
+import { useI18n } from 'vue-i18n'
+const { t, locale } = useI18n()
+
+onBeforeMount(() => {
+    for (const category in elementHierarchy) {
+        var categoryData = []
+        for (let i = 0; i < elementHierarchy[category].length; i++) {
+            const element = elementHierarchy[category][i]
+            if (!element.name.startsWith('verilog')) {
+                window.elementPanelList.push(element.label)
+                element['imgURL'] = getImgUrl(element.name)
+                categoryData.push(element)
+            }
+        }
+        panelData.push([category, categoryData])
+    }
+})
+
+function getImgUrl(elementName) {
+    const elementImg = new URL(
+        `../../../assets/img/${elementName}.svg`,
+        import.meta.url
+    ).href
+    return elementImg
+}
+
+var elementInput = ref('')
+function searchElements() {
+    if (!elementInput) return []
+    // logic to be imported from listener.js
+
+    const result = elementPanelList.filter((ele) =>
+        ele.toLowerCase().includes(elementInput.value)
+    )
+    var finalResult = []
+    for (const j in result) {
+        if (Object.prototype.hasOwnProperty.call(result, j)) {
+            for (const category in elementHierarchy) {
+                if (
+                    Object.prototype.hasOwnProperty.call(
+                        elementHierarchy,
+                        category
+                    )
+                ) {
+                    const categoryData = elementHierarchy[category]
+                    for (let i = 0; i < categoryData.length; i++) {
+                        if (result[j] == categoryData[i].label) {
+                            finalResult.push(categoryData[i])
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return finalResult
+}
+
+function createElement(elementName) {
+    if (simulationArea.lastSelected && simulationArea.lastSelected.newElement)
+        simulationArea.lastSelected.delete()
+    var obj = new modules[elementName]()
```

Comment:
lets enable `https://eslint.org/docs/latest/rules/no-var`, wdyt?

Reply:
If we add this rule we might have to disable it for the `simulators folder`, else it will add up to the linting errors since lots of places use var.
or insead i will use `let` and `const` from now on.